mods/default/functions.lua

```diff
@@ -715,6 +715,48 @@ function default.register_craft_metadata_copy(ingredient, result)
 	end)
 end
 
+--
+-- Log API / helpers
+--
+
+local log_non_player_actions = minetest.settings:get_bool("log_non_player_actions", false)
+
+-- `vector.check` is available since minetest >= 5.5.0
+local is_vector = vector.check or function(v)
+	return type(v) == "table" and
+		type(v.x) == "number" and type(v.y) == "number" and type(v.z) == "number"
+end
+
+function default.log_player_action(player, ...)
+	local msg = player:get_player_name()
+	if player.is_fake_player or not player:is_player() then
+		if not log_non_player_actions then
+			return
+		end
+		msg = msg .. "(" .. (type(player.is_fake_player) == "string"
+			and player.is_fake_player or "*").. ")"
+	end
+	for _, v in ipairs({...}) do
+		-- translate vectors
+		local part = is_vector(v) and minetest.pos_to_string(v) or v
+		-- no leading spaces before punctuation marks
+		msg = msg .. (string.match(part, "^[;,.]") and "" or " ") .. part
```

Comment:
I don't insist on removing this but since it's not needed inside MTG removing the string.match would simplify this

Reply:
If possible, i would like to keep it that way since i want to use this function in other mods in the future.